src/mame/bfm/bfm_cobra3.cpp

```diff
@@ -0,0 +1,592 @@
+
+
+l5380 -scsi controller
+
+ymz280b - in MAME - sound chip
+
+audio board
+
+75585116 - PALCE 16V8H
+75585118 - PALCE 16V8H
+
+TI introduce the TMS320AV110 MPEG audio decoder based on TI's 16 bits
+
+roms (RadioTimes)
+
+game 95 400 009 type swp left ver. radt 1.1 - ST 27C4001
+game 95 400 010 type swp right ver. radt 1.1 - ST 27C4001
+95 004 056 sound left - ST 27C4001
+95 004 057 sound right - AM 27C040
+
+cd
+
+version 95-100-302
+*/
+
+
+
+#include "emu.h"
+#include "bus/nscsi/cd.h"
+#include "machine/68340.h"
+#include "machine/meters.h"
+#include "machine/ncr5380.h"
+#include "machine/nscsi_bus.h"
+#include "machine/nvram.h"
+#include "video/ramdac.h"
+#include "machine/scc66470.h"
+#include "sound/ymz280b.h"
+#include "screen.h"
+#include "speaker.h"
+
+
+namespace {
+
+class bfm_cobra3_state : public driver_device
+{
+public:
+	bfm_cobra3_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag),
+			m_maincpu(*this, "maincpu"),
+			m_cpuregion(*this, "maincpu"),
+			m_nvram(*this, "nvram"),
+			m_ymz(*this, "ymz280b"),
+			m_palette(*this, "palette"),
+			m_ramdac(*this, "ramdac"),
+			m_scc66470(*this, "scc66470"),
+			// m_strobein(*this, "STROBE%u", 0),
+			m_meters(*this, "meters"),
+			m_lamps(*this, "lamp%u", 0U),
+			m_scsibus(*this, "scsi"),
+			m_scsic(*this, "scsi:6:ncr5380")
+	{ }
+
+	std::unique_ptr<uint16_t[]> m_mainram;
+
+	uint16_t bfm_cobra3_mem_r(offs_t offset, uint16_t mem_mask = ~0);
+	void bfm_cobra3_mem_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+
+	uint32_t screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
+
+	void bfm_cobra3(machine_config &config);
+	void ramdac_map(address_map &map);
+	void bfm_cobra3_map(address_map &map) ATTR_COLD;
+protected:
+
+	// devices
+	required_device<m68340_cpu_device> m_maincpu;
+	required_region_ptr<uint16_t> m_cpuregion;
+	required_device<nvram_device> m_nvram;
+	required_device<ymz280b_device> m_ymz;
+	required_device<palette_device> m_palette;
+	required_device<ramdac_device> m_ramdac;
+	required_device<scc66470_device> m_scc66470;
+	// required_ioport_array<4> m_strobein;
+	optional_device<meters_device> m_meters;
+	output_finder<256> m_lamps;
+	required_device<nscsi_bus_device> m_scsibus;
+	required_device<ncr5380_device> m_scsic;
+
+
+	virtual void machine_start() override ATTR_COLD;
+	static void scsi_devices(device_slot_interface &device);
+	void dma1_drq(int state);
+	void scc66470_map(address_map &map);
+	void scc66470_irq(int state);
+
+};
+
+uint16_t bfm_cobra3_state::bfm_cobra3_mem_r(offs_t offset, uint16_t mem_mask)
+{
+	int pc = m_maincpu->pc();
+	int cs = m_maincpu->get_cs(offset * 2);
+
+	switch ( cs )
+	{
+		case 1: //ROM
+			return m_cpuregion[offset & 0x7ffff];
+		case 2://RAM
+			return m_mainram[offset & 0x1fff];
+
+		case 3: //I/O 00ac0000-00ac07ff
+			{
+				offset &= 0x7ff;
+				offs_t cs_addr_8_11 = (offset * 2) & 0xf00;
+
+				switch(cs_addr_8_11)
```

Comment:
Please ensure consistent spacing, per https://docs.mamedev.org/contributing/cxx.html#spacing (compare to line 139).